# Prime In Numbers

This code solves [this puzzle](https://www.codewars.com/kata/54d512e62a5e54c96200019e), which basically asks for the prime factors of any number.

In [1]:
def primeFactors(n):
    p, c, t = set(), [], ''
    while n > 1:
        for i in range(2,int(n)+1):
            if n % i == 0:
                p.add(i)
                c += [i]
                n /= i
                break
    else:
        p = list(p)
        p.sort()
        e = [c.count(i) for i in p]
    for i in range(len(p)):
        t += '(%i**%i)' %(p[i],e[i])
    return t.replace('**1','')

In [2]:
print(primeFactors(100))
print(primeFactors(2019))
print(primeFactors(2525))
print(primeFactors(12))
print(primeFactors(365))

(2**2)(5**2)
(3)(673)
(5**2)(101)
(2**2)(3)
(5)(73)


These outputs are interpreted as:

$
100 = 2^2 \times 5^2 \\
2019 = 3 \times 673 \\
2525 = 5^2 \times 101 \\
12 = 2^2 \times 3 \\
365 = 5 \times 73
$

The prime factors are sorted in an ascending order, ignoring the powers.

The end.